test/zpool.test.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ */
+
+'use strict';
+
+var mod_kstat = require('kstat');
+var mod_bunyan = require('bunyan');
+var zpoolGZCollector = require('../lib/instrumenter/collectors-gz/zpool.js');
+
+var test = require('tape').test;
+
+var log = mod_bunyan.createLogger({
+    name: 'collector_test',
+    level: process.env['LOG_LEVEL'] || 'error',
+    stream: process.stderr
+});
+
+
+var kstatReader = new mod_kstat.Reader();
+var collector = new zpoolGZCollector({
+    log: log,
+    kstatReader: kstatReader
+});
+
+test('Validate zpool metrics', function _collectZpoolMetrics(t) {
+    var opts = {
+        zInfo: {
+            instanceId: 0,
+            zonename: 'gz'
+        }
+    };
+
+    collector.getMetrics(opts, function _gotZpoolMetrics(err, metrics) {
+        if (err) {
+            t.end(err);
+            return;
+        }
+
+        var valid_metric_types = ['counter', 'gauge', 'histogram'];
```

Comment:
It looks like the zpool metric code only uses 'gauge' type metrics at this point. Should we only include 'gauge' in this list? That might help to make sure we don't introduce backward-incompatible changes down the line.

Reply:
Good catch. We have two `counter` metrics `metaslab_group_loads` and `metaslab_group_unloads`, but we don't have histograms. I removed `histogram` from the list.

Reply:
Great, thanks for clarifying that.

Reply:
Thank you for taking the time to review this work.